stellarphot/settings/custom_widgets.py

```diff
@@ -0,0 +1,391 @@
+        self._choose_existing.options = choices
+
+    def _handle_selection(self, change):
+        if change["new"] == "none":
+
+            # We are making a new item...
+
+            # Hide the edit button
+            self._edit_button.layout.display = "none"
+
+            # This sets the ui back to its original state when created, i.e.
+            # everything is empty.
+            self._item_widget._init_ui()
+            self._item_widget.show_savebuttonbar = True
+            self._item_widget.disabled = False
+
+            # Set the validation status to invalid since the user must
+            # fill in the fields, and display the validation status
+            self._item_widget.is_valid.value = False
+            self._item_widget.is_valid.layout.display = "flex"
+
+            # Really only applies to PassbandMap, which has nested models,
+            # but does no harm in the other cases (true of both lines below)
+            # (and yes, both lines below are needed...this is a bug in ipyautoui,
+            #  I think, because open_nested=True isn't respected when we _init_ui.
+            #  Forcing a *change* in the value triggers the behavior we want.)
+            self._item_widget.open_nested = False
+            self._item_widget.open_nested = True
+
+        else:
+            # Display the selected item...
+            self._item_widget.show_savebuttonbar = False
+            self._item_widget.disabled = True
+            self._item_widget.is_valid.layout.display = "none"
+            self._item_widget.value = self._get_item(change["new"].name)
+
+            # Really only applies to PassbandMap, which has nested models,
+            # but does no harm in the other cases
+            self._set_disable_state_nested_models(self._item_widget, True)
+
+    def _edit_button_action(self, _):
+        """
+        Handle the edit button being clicked.
+        """
+        # Replace the display of the edit button with the save button bar...
+        self._edit_button.layout.display = "none"
+        self._item_widget.show_savebuttonbar = True
+        # ...enable the widget...
+        self._item_widget.disabled = False
+        # ...and show the validation status
+        self._item_widget.is_valid.layout.display = "flex"
+
+        # Enable the nested model components
+        self._set_disable_state_nested_models(self._item_widget, False)
+
+        # disable the name control, since the whole point of this
+        # is to be able to replace the values for a particular name
+        self._item_widget.di_widgets["name"].disabled = True
+
+        # Thie really only applies to PassbandMap, which has nested models,
+        # but does no harm in the other cases (true of both lines below)
+        # (and yes, both lines below are needed...this is a bug in ipyautoui,
+        #  I think, because open_nested=True isn't respected when we _init_ui.
+        #  Forcing a *change* in the value triggers the behavior we want.)
```

Comment:
Given you have the same comment and fix twice, makes me wonder if you should just create a function called `_jiggle_open_nested_for_ipyautoui`. :)